conformance/ion_literal.ion

```diff
@@ -0,0 +1,31 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+
+(ion_1_1
+  (each (toplevel '#$11'             '#$11'::'#$11'             '#$11'::1)
+        (toplevel $ion_literal $ion_literal::$ion_literal $ion_literal::1)
+        (produces $ion_literal               $ion_literal               1)))
+
+// But the annotation is only recognized at top level.
+(ion_1_1
+  (toplevel ['#$11',             '#$11'::'#$11',       $ion_literal::1])
+  (produces [$ion_literal, $ion_literal::$ion_literal, $ion_literal::1]))
+
+// No such effects in 1.0, even with the same SID.
+(ion_1_0
+  (toplevel $ion_symbol_table::{symbols:["$ion_literal"]})
+  (toplevel '#$10'             '#$10'::'#$10')
+  (produces $ion_literal $ion_literal::$ion_literal))
+
+
+// Top-level `$ion_literal::$ion_symbol_table::{}` is an error.
+// TODO Is this correct? What about '$ion_1_0' etc?
```

Comment:
Any IVM-like symbol with an annotation is already always a symbol, and doesn't need an annotation, right?

Also, how does this affect text Ion—specifically, a quoted, unannotated IVM is actually a no-op. Does resolving `$ion_literal` happen before or after determining whether it's a no-op?

Reply:
> Any IVM-like symbol with an annotation is already always a symbol, and doesn't need an annotation, right?

True, but that doesn't help when you want to represent the _unannotated_ symbol with text "$ion_1_0".

> Does resolving $ion_literal happen before or after determining whether it's a no-op?

They would have to happen at the same phase.  That no-op behavior of (eg) `'$ion_1_0'` means that either:
1.  the token is absorbed/discarded at the parsing level, as if it were a comment; or
2. the parser emits it as some new kind of syntactic entity (AST node type), to be discarded by the expansion phase; or
3. the parser emits it as a normal symbol, and the expansion phase discards it (at the same it would've detected any `$ion_literal` annotation).

I think (1) would be easiest to implement but surprising in terms of round-tripping. (2) is obnoxious. So I'm inclined to go with (2) above as a clearer specification of the 1.0 handling.

For 1.1 I currently think we should dump the no-op behavior of such **quoted** forms, making them normal symbols; let the unquoted form be "escaped" via `$ion_literal`, and do the same thing for local symtabs `$ion_literal::$ion_symbol_table::{`.  The outcome is that Ion 1.1 allows _everything_ in the data model to be written at top level as user data.

It wouldn't surprise me if Ion 1.0 implementations are full of bugs in their handling of this space.  We didn't think it through carefully enough, and then back-filled specification years later to try to match the observed behavior of the primary implementations.

This is worth a separate issue, especially since it's not even covered by this CR.  😉